### PR TITLE
Bug: Correctly merge custom CSS into global styles

### PR DESCRIPTION
**Pull request by Mamaduka**
Gutenberg issue: https://github.com/WordPress/gutenberg/issues/48466
Trac ticket: https://core.trac.wordpress.org/ticket/57833

### CODE REVIEW

File: src/wp-includes/block-editor.php [hunk 1]
```diff
@@ -418,7 +418,7 @@ function get_block_editor_settings( array $custom_settings, $block_editor_contex
 		 * Add the custom CSS as a separate stylesheet so any invalid CSS
 		 * entered by users does not break other global styles.
 		 */
-		$editor_settings['styles'][] = array(
```
Comment (Mamaduka) [inline src/wp-includes/block-editor.php:421]: See https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-editor.php#L439-L441.

**Comment by Mamaduka** [comment]
The test failure doesn't look related:

```
1) Tests_Admin_wpSiteHealth::test_object_cache_thresholds with data set #3 ('options_count', 100)
Failed asserting that false is true.
```

**Comment by gziolo** [comment]
The test passed after I re-run the failed job.

**Review by gziolo — APPROVED** [review]

**Comment by gziolo** [comment]
Committed with https://core.trac.wordpress.org/changeset/55438.

**Pull request closed without merging**